src/backend/utils/misc/queryenvironment.c

```diff
@@ -773,193 +684,62 @@ bool ENRgetSystableScan(Relation rel, Oid indexId, int nkeys, ScanKey key, List
 	return found;
 }
 
-static EphemeralNamedRelation
-find_enr(Form_pg_depend entry)
-{
-	QueryEnvironment *queryEnv = currentQueryEnv;
-	Oid catalog_oid = entry->classid;
-
-	ListCell         *curlc;
-
-	while (queryEnv)
-	{
-		switch (catalog_oid) {
-			/*
-			* pg_depend entry shows relation/type/constraint depends on a given object.
-			* Find the relation from ENR. If found, make sure
-			* to register the dependency of the ENR relation to this object.
-			*/
-			case RelationRelationId:
-				return get_ENR_withoid(queryEnv, entry->objid, ENR_TSQL_TEMP);
-
-			case TypeRelationId:
-				foreach(curlc, queryEnv->namedRelList) {
-					EphemeralNamedRelation tmp_enr;
-					ListCell *type_lc;
-
-					tmp_enr = (EphemeralNamedRelation) lfirst(curlc);
-					if (tmp_enr->md.enrtype != ENR_TSQL_TEMP)
-						continue;
 
-					foreach(type_lc, tmp_enr->md.cattups[ENR_CATTUP_TYPE])
-					{
-						Form_pg_type tup = ((Form_pg_type)GETSTRUCT((HeapTuple)lfirst(type_lc)));
-						if (tup->oid == entry->objid)
-							return tmp_enr;
-					}
-					foreach(type_lc, tmp_enr->md.cattups[ENR_CATTUP_ARRAYTYPE])
-					{
-						Form_pg_type tup = ((Form_pg_type)GETSTRUCT((HeapTuple)lfirst(type_lc)));
-						if (tup->oid == entry->objid)
-							return tmp_enr;
-					}
-				}
-				break;
+/*
+ * Add tuple to an ENR. It assumes that an ENR entry has been created with
+ * the relation name and relation oid.
+ */
+bool ENRAddTuple(Relation rel, HeapTuple tup)
```

Comment:
I have the same thought with Surendra, can we add more hook impl for functions like this one ?